lms/djangoapps/courseware/views.py

```diff
@@ -1370,3 +1371,33 @@ def _track_successful_certificate_generation(user_id, course_id):  # pylint: dis
                 }
             }
         )
+
+
+@require_http_methods(["GET", "POST"])
+def render_xblock(request, usage_key_string):
```

Comment:
Thanks for pointing this out, @ormsbee.  It was very useful in finding issues with the MongoDB queries.
I added a `bulk_operation` call, which reduced the queries from 16 to 6 (for Split).  And then changing the call from `get_module` to `get_module_for_descriptor` further reduced the count to 5.
There's an extra call to the database by the `edx_notes` decorator that would be nice to remove (later).

I've added calls to `check_mongo_calls` in the unit tests where appropriate.

I didn't add calls to verify the Sql queries since they seem to vary depending on whether the unit test is called singularly or within a bulk test of the whole file.  However, here's info from DjDT:
**Sql reads via index.html:** 19-23
**Sql reads via render_xblock endpoint:** 6-9